replica/database.hh

```diff
@@ -424,6 +424,11 @@ private:
     // Ensures that concurrent updates to sstable set will work correctly
     seastar::named_semaphore _sstable_set_mutation_sem = {1, named_semaphore_exception_factory{"sstable set mutation"}};
     mutable row_cache _cache; // Cache covers only sstables.
+    // FIXME: until we use uuid for sstables generation (#10459)
+    // _sstable_generation keeps track of the highest
+    // generation number in the table so that newly created sstables
+    // will use numeric generation numbers larger than this one.
+    // Initialized when the table is populated via update_sstables_known_generation.
     std::optional<int64_t> _sstable_generation = {};
```

Comment:
This should be replaced with a new_generation() method, that either creates a new random ID or uses _sstable_generation to pick up a new one. We shouldn't call operator++() on something that isn't an integer to do an operation that isn't incrementing something.

Reply:
is it something like #13073 ?

Reply:
Yes, but that looks like overkill to me.